packages/Filter/src/helpers/getIllogicalFilters.js

```diff
@@ -0,0 +1,30 @@
+import ruleIsBoolean from "./ruleIsBoolean";
+
+function putFiltersWithBooleanRulesLast(filterOne, filterTwo) {
+  const oneIsBinary = ruleIsBoolean(filterOne.rule);
+  const twoIsBinary = ruleIsBoolean(filterTwo.rule);
+
+  if (oneIsBinary === twoIsBinary) {
+    return 0;
+  }
+
+  return oneIsBinary ? 1 : -1;
+}
+
+// Get the illogical filters (use a boolean rule, and there are other filters on that column)
+export default function getIllogicalFilters(filters) {
+  const filtersClone = [...filters];
+  const processedColumnIds = [];
+
+  return filtersClone.sort(putFiltersWithBooleanRulesLast).filter(filter => {
+    if (processedColumnIds.includes(filter.columnId)) {
+      if (ruleIsBoolean(filter.rule)) {
+        return true;
+      }
+    } else {
+      processedColumnIds.push(filter.columnId);
+    }
+
+    return false;
```

Comment:
Can't understand the logic of this method. Why we do `processedColumnIds.push(filter.columnId)` and then return `false`?

Reply:
`return false` is within a `.filter()`, so it gets removed from the array (i.e. it is not an illogical filter).

The function first sorts all the the filters that exist (to simplify the following code), then goes over each.
If the current filter has a boolean rule (`=`, `!=`, etc) and is on a `column` that was already used by a filter, this one is illogical (so it will be removed).

For example, if the current one is `goals = 1` and there was already a filter on `goals`, `goals = 1` is illogical and will be removed.

Reply:
Oh, and the `.push()` is because we keep a tally of the `columnIds` that already have filters... so if we see another filter on that column, we know to look into it more deeply.